### ISSUE TITLE
Throw exception which takes constructor argument impossible

### ISSUE THREAD

**Issue by foens** [hafriedlander/phockito#11]
Hello there.

Suppose I want to mock some class, and that class needs to throw an exception which takes a constructor argument. This seems impossible with the current api.

It does also seem like Phockito is different from Mockito on this point. Mockito syntax would be:
Phockito->when($foo)->bar()->throw(new BazException("Argument here");

The only change needed is to replace
`else if ($response['action'] == 'throw') { $class = $response['value']; throw new $class(); }`
 with
`else if ($response['action'] == 'throw') { $class = $response['value']; throw $response['value']; }`

What do you think? This would break old tests, and therefore should probably use some other method to register the throwable class. Maybe `->throwInstance(new BazException("foo"))`

**Comment by hafriedlander** [comment]
Good point, although it should be possible to work with either call, i.e.

``` php
else if ($response['action'] == 'throw') { 
  $class = $response['value']; 
  throw (is_object($class) ? $class : new $class()) ;
}
```

**Comment by foens** [comment]
That sounds even better :) Gogo

**Comment by hafriedlander** [comment]
Fixed in fe7ac22c4cff843d6edcc5b0acccabe7a73b508a
